zrml/prediction-markets/src/migrations.rs

```diff
@@ -15,259 +15,328 @@
+                oracle: 3,
+                metadata: vec![4, 5],
+                market_type: MarketType::Scalar(14..=15),
+                period: MarketPeriod::Block(6..7),
+                deadlines: Deadlines { grace_period: 8, oracle_duration: 9, dispute_duration: 10 },
+                scoring_rule: ScoringRule::CPMM,
+                status: MarketStatus::Resolved,
+                report: Some(Report { at: 11, by: 12, outcome: OutcomeReport::Categorical(13) }),
+                resolved_outcome: Some(OutcomeReport::Categorical(13)),
+                dispute_mechanism: MarketDisputeMechanism::Court,
+            };
+            insert_market::<Blake2_128Concat>(market_id, market.clone());
 
-        let market_ids_per_report_iterator: IterType<T> =
-            frame_support::migration::storage_key_iter::<_, _, Twox64Concat>(
-                b"PredictionMarkets",
-                b"MarketIdsPerReportBlock",
+            let dispute =
+                MarketDisputeOf::<Runtime> { at: 16, by: 17, outcome: OutcomeReport::Scalar(18) };
+            Disputes::<Runtime>::insert(
+                market_id,
+                BoundedVec::try_from(vec![dispute.clone()]).unwrap(),
             );
 
-        let market_ids_per_report: Vec<DisputeBlockToMarketIdsTuple<T>> =
-            market_ids_per_report_iterator.collect();
-        total_weight = total_weight
-            .saturating_add(T::DbWeight::get().reads(market_ids_per_report.len() as u64));
-        let mut updated_data = Vec::new();
-        for (dispute_start_block, market_ids) in market_ids_per_report {
-            let dispute_duration = T::DisputePeriod::get();
-            let new_dispute_start_block = dispute_start_block.saturating_add(dispute_duration);
-            updated_data.push((new_dispute_start_block, market_ids));
-            MarketIdsPerReportBlock::<T>::remove(dispute_start_block);
-            total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
-        }
-        for (new_dispute_start_block, market_ids) in updated_data {
-            total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
-            MarketIdsPerReportBlock::<T>::insert(new_dispute_start_block, market_ids);
-        }
-        StorageVersion::new(PREDICTION_MARKETS_NEXT_STORAGE_VERSION_FOR_MIGRATE_MARKET_IDS_STORAGE)
-            .put::<Pallet<T>>();
-        total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+            TransformScalarMarketsToFixedPoint::<Runtime>::on_runtime_upgrade();
 
-        log::info!("Completed storage migration of MigrateMarketIdsPerBlockStorage");
-        total_weight
+            let market_after = Markets::<Runtime>::get(market_id).unwrap();
+            assert_eq!(market_after, market);
+
+            let disputes_after = Disputes::<Runtime>::get(market_id);
+            assert_eq!(disputes_after.len(), 1);
+            assert_eq!(disputes_after[0], dispute);
+        });
```

Comment:
I don't really get it. Where do you skip what and how here?

Reply:
Good lord, that diff... 😐

Yeah, that test name is terrible. What I'm testing here is that the migration is noop if the version numbers are incorrect.

Reply:
Just a dumb question: Where do you set the version numbers in this test case?

Reply:
Because you did not use `set_up_chain();` ?

Reply:
That's hard to understand, but because it's only a transitory test it's okay.

Reply:
Yeah, the point is that `setup_chain` creates a situation in which the update is applied. By default, the pallet versions are so that the update will not trigger, so not using `setup_chains` means that update doesn't happen. I've made the test a little clearer, hope that helps.